### PR TITLE
Update `CHANGELOG.md` for JavaScriptKit 0.10.0

### PR DESCRIPTION
**Pull request by MaxDesiatov**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,33 @@
...
+**Merged pull requests:**
+
+- Simplify `JSPromise` API ([#115](https://github.com/swiftwasm/JavaScriptKit/pull/115)) via [@kateinoigakukun](https://github.com/kateinoigakukun)
+- Create `FUNDING.yml` ([#117](https://github.com/swiftwasm/JavaScriptKit/pull/117)) via [@MaxDesiatov](https://github.com/MaxDesiatov)
+- Major API change on `JSClosure` ([#113](https://github.com/swiftwasm/JavaScriptKit/pull/113)) via [@kateinoigakukun](https://github.com/kateinoigakukun)
+- Update `package.json` to lockfileVersion 2 ([#114](https://github.com/swiftwasm/JavaScriptKit/pull/114)) via [@kateinoigakukun](https://github.com/kateinoigakukun)
+- Bump `ini` from 1.3.5 to 1.3.8 in `/Example` ([#111](https://github.com/swiftwasm/JavaScriptKit/pull/111)) via [@dependabot[bot]](https://github.com/dependabot[bot])
+- Update doc comment in `JSTypedArray.swift` ([#110](https://github.com/swiftwasm/JavaScriptKit/pull/110)) via [@MaxDesiatov](https://github.com/MaxDesiatov)
+
+
```
Comment (MaxDesiatov) [inline CHANGELOG.md:30]:
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,32 @@
+# 0.10.0 (21 January 2021)
+
+This release contains multiple breaking changes in preparation for enabling `async`/`await`, when
+this feature is available in a stable SwiftWasm release. Namely:
+
+* `JSClosure.init(_ body: @escaping ([JSValue]) -> ())` overload is deprecated to simplify type
+checking. Its presence requires explicit type signatures at the place of use. It will be removed
+in the future version of JavaScriptKit.
```
Comment (j-f1) [inline CHANGELOG.md:8]:
```suggestion
in the next version of JavaScriptKit.
```

Or, if not the next version:
```suggestion
in a future version of JavaScriptKit.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,32 @@
...
+* `JSClosure` is no longer a subclass of `JSFunction`. These classes are not related enough to keep
+them in the same class hierarchy.
+* Introduced `JSOneshotClosure` for closures that are going to be called only once. You don't need
+to manage references to these closures manually, as opposed to `JSClosure`.
```
Comment (j-f1) [inline CHANGELOG.md:12]:
```suggestion
to manage references to these closures manually, as opposed to `JSClosure`.
However, they can only be called a single time from the JS side. Subsequent invocation attempts will raise a fatal error on the Swift side.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,32 @@
+# 0.10.0 (21 January 2021)
+
+This release contains multiple breaking changes in preparation for enabling `async`/`await`, when
+this feature is available in a stable SwiftWasm release. Namely:
+
+* `JSClosure.init(_ body: @escaping ([JSValue]) -> ())` overload is deprecated to simplify type
+checking. Its presence requires explicit type signatures at the place of use. It will be removed
+in the future version of JavaScriptKit.
+* `JSClosure` is no longer a subclass of `JSFunction`. These classes are not related enough to keep
+them in the same class hierarchy.
```
Comment (j-f1) [inline CHANGELOG.md:10]:
```suggestion
them in the same class hierarchy.
As a result, you can no longer call `JSClosure` objects directly from Swift.
```

File: CHANGELOG.md (hunk 2 above)
Comment (MaxDesiatov) [reply CHANGELOG.md:8]: All good points, thanks!

**Review by j-f1 — APPROVED** [review]

**Pull request merged**